doc/OpenMPSupport.md

```diff
@@ -0,0 +1,220 @@
+Additionally, the ```__kmpc_fork_call``` receives the following arguments:
+- Two parameters for internal OpenMP configurations.
+- The reference to the ```.omp_outlined.```, where the actual logic is.
+- A dynamic number of variables that corresponds to the reference to the shared variables.
+
+### Taken approach
+To bypass the runtime library functions, it is necessary to let TAFFO understand what functions will be called.
+
+Although the integration started with the goal to instruct every pass to bypass the indirect function and analyze the outlined one,
+the path was discarded since substantial modifications were required in every pass of TAFFO.
+ 
+A simpler but more effective approach is currently implemented, with modifications only to the Initialization and Conversion passes.
+Instead of resolving the indirections at each pass, the library functions are replaced with the outlined functions during the Initialization pass. 
+Finally, during the last pass, the library function calls are re-inserted, accounting for the new fixed-point functions created by TAFFO.
+
+The chosen approach revolves around so-called "trampolines". A trampoline is a custom function that directly calls an indirect function. 
+This is convenient because it allows TAFFO to analyze the indirect calls as if they were direct calls, which are already supported.
+
+#### Initialization pass
+From the above example of the `@__kmpc_fork_call`, a corresponding `@__kmpc_fork_call_trampoline` is created and inserted in place of the original one. 
+The body of this function is very simple and contains:
+- the direct call to the outlined function, with the related arguments (the shared variables in OpenMP).
+- an additional instruction that is used to preserve information about the original function. 
+ In particular, the original function is saved in a local variable using a bitcast instruction. 
+ This is necessary because otherwise the following optimization passes, such as Dead Code Elimination (DCE), 
+ would remove the unused original function, making it impossible to reinsert the original calls in the Conversion pass.
+
+#### Range Analysis and Data Type Allocation passes
+TAFFO will regularly analyze the trampoline calls,  without any modification to these intermediate passes.
+
+#### Conversion pass
+At the end of the Conversion pass, it is necessary to transform back the trampolines into their original calls. 
+To do so, the original function is picked from the saved bitcast and placed in place of the trampoline call, which will be deleted. 
+The original function will have the relevant arguments replaced accordingly, specifically the indirect function to be called and the shared variables needed. 
+
+In the above example, TAFFO will correctly convert the floating-point instructions in fixed-points, and
+the call to `@__kmpc_fork_call` will contain `@.omp_outlined..2_fixp` instead of `@.omp_outlined.` and `i32* %foo.s6_26fixp` instead of `float* %foo`.
+
+## Benchmarks
```

Comment:
The benchmark evaluation should go into a new document.
It is more of a one-time thing, rather than a document we want to keep up to date

Reply:
Moved in another document